Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1

```diff
@@ -606,13 +606,52 @@ function InvokeDTAExecHostExe([string] $Version, [System.Management.Automation.P
     }
     $exePath = Join-Path -Path $vsRoot -ChildPath $ExeName
     $exePath = "'" + $exePath + "'"
+	Try
+	{
+		$session = CreateNewSession -MachineCredential $MachineCredential
+		Invoke-Command -Session $session -ErrorAction Continue -ErrorVariable err -OutVariable out -scriptBlock { schtasks.exe /create /TN:DTAConfig /TR:$args /F /RL:HIGHEST /SD:01/01/2050 /SC:ONCE /ST:00:00 ; schtasks.exe /run /TN:DTAConfig } -ArgumentList $exePath
 
-    $session = New-PSSession -ComputerName . -Credential $MachineCredential
-    Invoke-Command -Session $session -ErrorAction Continue -ErrorVariable err -OutVariable out -scriptBlock { schtasks.exe /create /TN:DTAConfig /TR:$args /F /RL:HIGHEST /SD:01/01/2050 /SC:ONCE /ST:00:00 ; schtasks.exe /run /TN:DTAConfig } -ArgumentList $exePath
+		Write-Verbose ("Error : {0} " -f ($err | out-string)) -Verbose
+		Write-Verbose ("Output : {0} " -f ($out | out-string)) -Verbose
+	}
+    Catch
+    {
+        Write-Verbose -Message ("Unable to start Agent process, will be rebooting the machine to complete the configuration") -Verbose
+    }
+   
+}
 
-    Write-Verbose ("Error : {0} " -f ($err | out-string)) -Verbose
-    Write-Verbose ("Output : {0} " -f ($out | out-string)) -Verbose
+function CreateNewSession( [System.Management.Automation.PSCredential] $MachineCredentials)
+{
+    $winrmconfigDetails = Winrm e  winrm/config/listener -format:pretty |Out-String
+    $xmldoc = [XML]$winrmconfigDetails    
+    $transports = $xmldoc.GetElementsByTagName("cfg:Transport")
+    $ports = $xmldoc.GetElementsByTagName("cfg:Port")
+   
+    if( $ports -ne $null -and  $transports -ne $null -and $ports.Count -gt 0)
+    {
+        $port = $ports[0].InnerText
+        $transport = $transports[0].InnerText
+    }
+    else
+    {
+        Write-Verbose -Message("Unable to fetch WinRM config details. Using default port for configuration") -Verbose
+        $port = 5985
```

Comment:
is this hardcoding intentional ?
scenario will break if anyone is using a non default port